java/bulk-export/bulk-export-planner/src/main/java/sleeper/bulkexport/planner/SqsBulkExportProcessorLambda.java

```diff
@@ -86,28 +106,53 @@ public Void handleRequest(SQSEvent event, Context context) {
         event.getRecords().stream()
                 .map(SQSEvent.SQSMessage::getBody)
                 .peek(body -> LOGGER.debug("Received message with body {}", body))
-                .flatMap(body -> deserialiseAndValidate(body).stream())
+                .flatMap(body -> {
+                    try {
+                        return deserialiseAndValidate(body).stream();
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                })
                 .forEach(query -> {
                     try {
                         processor.processExport(query);
                     } catch (ObjectFactoryException e) {
                         LOGGER.error("Failed to process export query", e);
+                        sendToDeadLetterQueue(query.toString());
+                        throw new RuntimeException(e);
                     }
                 });
         return null;
     }
 
-    private Optional<BulkExportQuery> deserialiseAndValidate(String message) {
+    private Optional<BulkExportQuery> deserialiseAndValidate(String message) throws Exception {
         try {
             BulkExportQuery exportQuery = bulkExportQuerySerDe.fromJson(message);
             LOGGER.debug("Deserialised message to export query {}", exportQuery);
             return Optional.of(exportQuery);
         } catch (BulkExportQueryValidationException e) {
             LOGGER.error("QueryValidationException validating export query from JSON {}", message, e);
-            return Optional.empty();
+            sendToDeadLetterQueue(message);
+            throw e;
         } catch (RuntimeException e) {
             LOGGER.error("Failed deserialising export query from JSON {}", message, e);
-            return Optional.empty();
+            sendToDeadLetterQueue(message);
+            throw e;
```

Comment:
The call to sendToDeadLetterQueue is not needed. Throwing the exception all the way out will already result in the message going to the dead letter queue, because the queue is configured to send any unsuccessful message to the dead letter queue.

This code will result in two dead letters.